java/code/src/com/redhat/rhn/manager/recurringactions/RecurringActionManager.java

```diff
@@ -0,0 +1,137 @@
+import com.redhat.rhn.domain.recurringactions.OrgRecurringAction;
+import com.redhat.rhn.domain.recurringactions.RecurringActionFactory;
+import com.redhat.rhn.domain.role.RoleFactory;
+import com.redhat.rhn.domain.server.MinionServer;
+import com.redhat.rhn.domain.server.MinionServerFactory;
+import com.redhat.rhn.domain.user.User;
+import com.redhat.rhn.manager.EntityNotExistsException;
+import com.redhat.rhn.manager.system.ServerGroupManager;
+import com.redhat.rhn.manager.system.SystemManager;
+import com.redhat.rhn.taskomatic.TaskomaticApi;
+import com.redhat.rhn.taskomatic.TaskomaticApiException;
+
+import java.util.List;
+
+/**
+ * RecurringActionManager
+ * @version $Rev$
+ */
+public class RecurringActionManager {
+
+    private static  TaskomaticApi taskomaticApi = new TaskomaticApi();
+    private static final RecurringActionManager MANAGER = new RecurringActionManager();
+
+    /**
+     * Set the {@link TaskomaticApi} instance to use. Only needed for unit tests.
+     * @param taskomaticApiIn the {@link TaskomaticApi}
+     */
+    public static void setTaskomaticApi(TaskomaticApi taskomaticApiIn) {
+        taskomaticApi = taskomaticApiIn;
+    }
+    /**
+     * Singleton Instance to get manager object
+     * @return an instance of the manager
+     */
+    public static RecurringActionManager getInstance() {
+        return MANAGER;
+    }
+
+    /**
+     * Private constructor.
+     */
+    private RecurringActionManager() {
+    }
+
+    /**
+     * Create a new minion recurring action
+     *
+     * @param minionId id of the minion
+     * @param cron the cron expression
+     * @param testMode if test mode
+     * @param active is is active
+     * @param user the user
+     * @throws TaskomaticApiException if TaskoSchedule creation fails
+     */
+    public static void createMinionRecurringAction(long minionId, String cron, boolean testMode,
+                                                   boolean active, User user) throws TaskomaticApiException {
+        if (!SystemManager.isAvailableToUser(user, minionId)) {
+            throw new PermissionException("Minion not accessible to user");
+        }
+        MinionServer minion = MinionServerFactory.lookupById(minionId)
+                .orElseThrow(() -> new EntityNotExistsException(MinionServer.class, minionId));
+        MinionRecurringAction action = new MinionRecurringAction(testMode, active, minion);
+        RecurringActionFactory.save(action);
+
+        taskomaticApi.scheduleSatBunch(user, action.computeTaskoScheduleName(), "recurring-state-apply-bunch", cron);
+    }
+
+    /**
+     * List minion recurring action with minion id
+     *
+     * @param minionId id of the minion
+     * @param user the user
+     * @return list of minion recurring actions
+     */
+    public static List<MinionRecurringAction> listMinionRecurringAction(long minionId, User user) {
+        if (!SystemManager.isAvailableToUser(user, minionId)) {
```

Comment:
Suggestion: instead of `isAvailableUser`, we could use `ensureAvailableToUser`. With this we'll get a more detailed exception info:

```java
        try {
            SystemManager.ensureAvailableToUser(user, minionId);
        }
        catch (LookupException e) {
            throw new PermissionException("Minion not accessible to user", e);
        }
```